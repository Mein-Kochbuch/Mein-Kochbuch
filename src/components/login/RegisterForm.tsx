import React, {useState} from "react";
import {Linking} from "react-native";
import InputField from "./InputField";
import styled from "styled-components/native";
import {Colors} from "react-native/Libraries/NewAppScreen";
import {useColorScheme} from "react-native-appearance";
import Validator from "../../utils/Validator";

interface RegisterFormProps {
    handleRegister: (email: string, username: string, password: string) => void
}

export default function RegisterForm({handleRegister}: RegisterFormProps) {
    const [email, setEmail] = useState<string>("")
    const [name, setName] = useState<string>("")
    const [password, setPassword] = useState<string>("")
    const [passwordDuplicate, setPasswordDuplicate] = useState<string>("")

    const [passwordError, setPasswordError] = useState<string | undefined>()
    const [emailError, setEmailError] = useState<string | undefined>()

    const isDarkMode = useColorScheme() === 'dark';
    const {validatePassword, validateEmail} = Validator()

    const handleSubmit = () => {
        const passwordValidationError = validatePassword(password, passwordDuplicate)
        if (passwordValidationError) {
            setPasswordError(passwordValidationError)
            return
        }
        setPasswordError("")

        const emailValidationError = validateEmail(email)
        if (emailValidationError) {
            setEmailError(emailValidationError)
            return
        }
        setEmailError("")

        handleRegister(email, name, password)
    }

    return (
        <StyledView>
            <InputField
                title={"E-Mail"}
                errorText={emailError}
                onChangeText={setEmail}
                placeholder="E-Mail"
                value={email}
                keyboardType={"email-address"}/>
            <InputField
                title={"Name"}
                onChangeText={setName}
                placeholder="Name"
                value={name}/>
            <InputField
                title={"Password"}
                onChangeText={setPassword}
                placeholder="Password"
                value={password}
                secureTextEntry={true}/>
            <InputField
                title={"Password"}
                errorText={passwordError}
                onChangeText={setPasswordDuplicate}
                placeholder="Password"
                value={passwordDuplicate}
                secureTextEntry={true}/>

            <PrivacyBanner>
                Wenn du fortfährst, stimmst du den
                <LinkText onPress={() => Linking.openURL('https://mein-kochbuch.org/agb/')}> AGBs </LinkText>
                zu. In unseren
                <LinkText onPress={() => Linking.openURL('https://mein-kochbuch.org/privacypolicy/')}> Privacy
                    Policy </LinkText>
                erfährst du, wie wir deine Daten erfassen, verwenden und teilen.
            </PrivacyBanner>

            <StyledTouchableOpacity isDarkMode={isDarkMode} onPress={handleSubmit}>
                <StyledText>
                    Register
                </StyledText>
            </StyledTouchableOpacity>
        </StyledView>
    )
}

interface StyledTouchableOpacityProps {
    isDarkMode: boolean
}

const StyledView = styled.View`
  display: flex;
  align-items: center;
  flex-direction: column;
`

const LinkText = styled.Text`
  color: #0074D9;
  text-decoration-line: underline;
`

const PrivacyBanner = styled.Text`
  margin: 24px;
  text-align: center;
`

const StyledText = styled.Text`
  text-align: center;
  font-size: 30px;
`

const StyledTouchableOpacity = styled.TouchableOpacity<StyledTouchableOpacityProps>`
  margin: 12px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  width: 80%;
  min-width: 300px;
  border-style: solid;
  border-color: ${props => props.isDarkMode ? Colors.lighter : Colors.darker};
  border-radius: 10px;
  border-width: 2px;
`

