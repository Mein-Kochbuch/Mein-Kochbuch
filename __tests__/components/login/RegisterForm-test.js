import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import RegisterForm from "../../../src/components/login/RegisterForm";

jest.mock("react-native-appearance", () => ({
    useColorScheme: jest.fn(),
}))

it('RegisterForm Test', () => {

    const handleRegister = () => {
    }

    const component = renderer.create(
        <RegisterForm handleRegister={handleRegister}/>
    );

    expect(component.toJSON()).toMatchSnapshot()
})
