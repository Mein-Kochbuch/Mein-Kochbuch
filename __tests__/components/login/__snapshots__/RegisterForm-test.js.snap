// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm Test 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 12,
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 12,
        },
      ]
    }
  >
    <Text
      style={Object {}}
    >
      E-Mail
    </Text>
    <TextInput
      isDarkMode={false}
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="E-Mail"
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#222",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 2,
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "minWidth": 300,
            "width": "80%",
          },
        ]
      }
      value=""
    />
  </View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 12,
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 12,
        },
      ]
    }
  >
    <Text
      style={Object {}}
    >
      Name
    </Text>
    <TextInput
      isDarkMode={false}
      onChangeText={[Function]}
      placeholder="Name"
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#222",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 2,
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "minWidth": 300,
            "width": "80%",
          },
        ]
      }
      value=""
    />
  </View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 12,
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 12,
        },
      ]
    }
  >
    <Text
      style={Object {}}
    >
      Password
    </Text>
    <TextInput
      isDarkMode={false}
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#222",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 2,
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "minWidth": 300,
            "width": "80%",
          },
        ]
      }
      value=""
    />
  </View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 12,
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 12,
        },
      ]
    }
  >
    <Text
      style={Object {}}
    >
      Password
    </Text>
    <TextInput
      isDarkMode={false}
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#222",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 2,
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "minWidth": 300,
            "width": "80%",
          },
        ]
      }
      value=""
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "textAlign": "center",
        },
      ]
    }
  >
    Wenn du fortfährst, stimmst du den
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#0074D9",
            "textDecorationLine": "underline",
          },
        ]
      }
    >
       AGBs 
    </Text>
    zu. In unseren
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#0074D9",
            "textDecorationLine": "underline",
          },
        ]
      }
    >
       Privacy Policy 
    </Text>
    erfährst du, wie wir deine Daten erfassen, verwenden und teilen.
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#222",
        "borderRadius": 10,
        "borderStyle": "solid",
        "borderWidth": 2,
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "space-between",
        "marginBottom": 12,
        "marginLeft": 12,
        "marginRight": 12,
        "marginTop": 12,
        "minWidth": 300,
        "opacity": 1,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 30,
            "textAlign": "center",
          },
        ]
      }
    >
      Register
    </Text>
  </View>
</View>
`;
